File: leetcode/rookie/07_两数之和.js
const twoSum = function (nums = [], target) {
  const map = new Map();
  for (let i = 0; i < nums.length; i++) {
    const otherIndex = map.get(target - nums[i]);
    if (otherIndex !== undefined) {
      return [otherIndex, i];
    }
    map.set(nums[i], i);
  }
};

console.log(twoSum([3,2,5], 8))

File: leetcode/rookie/07_两数之和.py
def twoSum(nums, target):
    hashmap = {}
    for i, num in enumerate(nums):
        j = hashmap.get(target - num)
        if j is not None:
            return [j, i]
        hashmap[num] = i


a = twoSum([3, 2, 5], 8)

print(a)
